sirepo_bluesky/sirepo_bluesky.py

```diff
@@ -112,18 +112,15 @@ def find_optic_id_by_name(self, optic_name):
                 return optic_id
         raise ValueError(f'Not valid optic {optic_name}')
 
-    def get_datafile(self):
+    def get_datafile(self, file_index=-1):
         """ Request the raw datafile of simulation results from the server.
 
             Notes
             -----
             Call auth() and run_simulation() before this.
         """
         assert hasattr(self, 'cookies'), 'call auth() before get_datafile()'
-        if self.sim_type in ["srw", "shadow"]:
-            url = 'download-data-file/{}/{}/{}/-1'.format(self.sim_type, self.sim_id, self.data['report'])
-        elif self.sim_type in ["madx"]:
-            url = 'download-data-file/{}/{}/{}/0'.format(self.sim_type, self.sim_id, self.data['report'])
+        url = f"download-data-file/{self.sim_type}/{self.sim_id}/{self.data['report']}/{file_index}"
```

Comment:
Probably we need to address the case of `file_index=0` in the relevant code that uses this method. Could you please have a look?